community/kernel/src/main/java/org/neo4j/kernel/impl/nioneo/store/DumpStore.java

```diff
@@ -175,43 +192,72 @@ public final void dump( STORE store ) throws Exception
         out.println( "store.getRecordSize() = " + size );
         out.println( "<dump>" );
         long used = 0;
-        for ( long i = 1, high = store.getHighestPossibleIdInUse(); i <= high; i++ )
+
+        if ( ids == null )
+        {
+            long high = store.getHighestPossibleIdInUse();
+
+            for ( long id = 1; id <= high; id++ )
+            {
+                boolean inUse = dumpRecord( store, size, fileChannel, buffer, id );
+
+                if ( inUse )
+                {
+                    used++;
+                }
+            }
+        }
+        else
+        {
+            for( long id : ids )
+            {
+                dumpRecord( store, size, fileChannel, buffer, id );
+            }
+        }
+        out.println( "</dump>" );
+
+        if ( ids == null )
+        {
+            out.printf( "used = %s / highId = %s (%.2f%%)%n", used, store.getHighId(), used * 100.0 / store.getHighId
+                    () );
+
+        }
+    }
+
+    private boolean dumpRecord( STORE store, int size, StoreChannel fileChannel, ByteBuffer buffer, long id ) throws
+            Exception
+    {
+        RECORD record = store.forceGetRecord( id );
+
+        Object transform = transform( record );
+        if ( transform != null )
+        {
+            if ( !"".equals( transform ) )
+            {
+                out.println( transform );
+            }
+        }
+        else
         {
-            RECORD record = store.forceGetRecord( i );
+            out.print( record );
+            buffer.clear();
+            fileChannel.read( buffer, id * size );
+            buffer.flip();
             if ( record.inUse() )
             {
-                used++;
+                dumpHex( buffer, id * size );
             }
-            Object transform = transform( record );
-            if ( transform != null )
+            else if ( allZero( buffer ) )
```

Comment:
I like this idea, but hopefully this path is rarely taken. This is the "raw" format when there is no suitable transform function.